ksqldb-engine/src/main/java/io/confluent/ksql/util/SharedKafkaStreamsRuntimeImpl.java

```diff
@@ -247,6 +248,8 @@ public void start(final QueryId queryId) {
 
   @Override
   public void overrideStreamsProperties(final Map<String, Object> newProperties) {
+    newProperties.put(StreamsConfig.APPLICATION_SERVER_CONFIG,
+        streamsProperties.get(StreamsConfig.APPLICATION_SERVER_CONFIG));
```

Comment:
this could benefit from a comment, what could cause `newProperties` to have a different `APPLITCATION_SERVER_CONFIG` value and can we just prevent that from happening in first place? this seems like it silently overwrites something that the user wanted to do and can cause some confusion